torchtnt/framework/callbacks/_checkpoint_utils.py

```diff
@@ -22,7 +22,36 @@
 
 logger: logging.Logger = logging.getLogger(__name__)
 
+T = TypeVar("T")
 
+
+def rank_zero_read_and_broadcast(
```

Comment:
This could be part of utils/distributed instead. it's general enough, though getting the process group from kwargs needs to be better documented

Reply:
Good point, and it doesn't have to be limited to just reads, essentially any task can be done with this. Will move